dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerTest.kt

```diff
@@ -402,4 +420,182 @@ class CompanyDataControllerTest {
             "Dataland does not contain the posted company.",
         )
     }
+
+    /*@Test
+    fun `check that an exception is thrown if the contactDetails does not have the valid format`() {
+        val uploadInfo = apiAccessor.uploadNCompaniesWithoutIdentifiers(1).first()
+        val companyId = uploadInfo.actualStoredCompany.companyId
+        val patchObject = CompanyInformationPatch(
+            companyContactDetails = setOf("New-Email-1"),
+        )
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        apiAccessor.companyDataControllerApi.patchCompanyById(
+            companyId,
+            patchObject,
+        )
+
+        val exception = assertThrows<ClientException> {
+            apiAccessor.companyDataControllerApi.patchCompanyById(
+                companyId,
+                patchObject,
+            )
+        }
+        assertTrue(exception.statusCode == 403, "The exception should indicate unauthorized access (HTTP 403)")
+    }
+*/
+    @Test
+    fun `check that the dataUploader can patch contactDetails and website if the company does not have companyOwner`() {
+        val uploadInfo = apiAccessor.uploadNCompaniesWithoutIdentifiers(1).first()
+        val companyId = uploadInfo.actualStoredCompany.companyId
+        val patchObject = CompanyInformationPatch(
+            companyContactDetails = setOf("New-Email-1@example.com", "New-Email-2@example.com"),
+            website = "New-Website",
+        )
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+        val updatedCompany = apiAccessor.companyDataControllerApi.patchCompanyById(
+            companyId,
+            patchObject,
+        )
+        assertEquals(
+            patchObject.companyContactDetails!!, updatedCompany.companyInformation.companyContactDetails,
+            "The company contact details should have been updated",
+        )
+        assertEquals(
+            patchObject.website!!, updatedCompany.companyInformation.website,
+            "The company website should have been updated",
+        )
+    }
+
+    @Test
+    fun `check that the dataUploader cannot patch contactDetails or website if the company has a companyOwner`() {
```

Comment:
schreib lieber "check that Dataland uploader can patch ..."

dann ist klar, dass du das für einen user testest, der die Dataland Uploader Rolle hat

"dataUploader" versteht man nur, wenn man weiß, dass das ein technical account ist

Reply:
done